include/onnxruntime/core/session/onnxruntime_c_api.h

```diff
@@ -143,15 +143,6 @@ typedef enum OrtErrorCode {
   ORT_EP_FAIL,
 } OrtErrorCode;
 
-// This configures the arena based allocator used by ORT
-// See docs/C_API.md for details on what these mean and how to choose these values
-typedef struct OrtArenaCfg {
```

Comment:
Removed this definition from the header (and made this struct opaque to the user) as we now have an API to create an instance and we want to remove ambiguity as to how to instantiate this struct. 

This change preserved ABI but introduces source incompatibility which is okay because we don't guarantee source compatibility.